src/rules/no_await_in_loop.rs

```diff
@@ -88,6 +114,66 @@ impl<'c> NoAwaitInLoopVisitor<'c> {
   }
 }
 
+impl<'c> Handler for NoAwaitInLoopVisitor<'c> {
+  fn await_expr(&mut self, await_expr: &AstView::AwaitExpr) {
+    fn inside_loop(
+      await_expr: &AstView::AwaitExpr,
+      node: AstView::Node,
+    ) -> bool {
+      use NodeKind::*;
+      match node.kind() {
```

Comment:
Instead of matching on `node.kind()` and doing `.expect`, it's possible to match directly on the node.

Additionally, instead of doing `.inner.await_token` it's possible to do `.await_token()` instead, which will get `inner.await_token()` (there should be a helper method for ever property found on `inner`, so `.inner.x` -> `.x()`).

For example:

```ts
use Node::*;
match node {
  FnDecl(_) | FnExpr(_) | ArrowExpr(_) => false,
  ForOfStmt(stmt)
    if stmt.await_token().is_some() =>
  {
    // etc...
  }
  // etc...
}
```

Reply:
Thanks! Sounds good suggestion, I rewrote the code.

Reply:
Looks really good!